src/CSV.jl

```diff
@@ -61,6 +87,10 @@ function Base.show(io::IO, f::File)
     show(io, Tables.schema(f))
 end
 
+Base.IndexStyle(::Type{File}) = Base.IndexLinear()
+Base.eltype(f::File{threaded}) where {threaded} = Row{threaded}
+Base.size(f::File) = (getrows(f),)
+
```

Comment:
I think one usually defines `Base.axes` also?

```
Base.axes(f::File) = (Base.OneTo(getrows(f)),)
```

Reply:
I suppose it would make sense to place the `getindex` implementation here since it's a part of the `AbstractArray` interface, keeping it together.

Reply:
That's the default definition, so I think we're good.

Reply:
Hmm... right you are! Though I wonder if `Base` has it backwards - `size` could derive from `axes` without making any assumptions.